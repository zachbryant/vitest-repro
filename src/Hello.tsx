import * as React from "react";

import Def from "./def";

export const Hello: React.FC = () => {
  return (
    <div>
      <Def />
    </div>
  );
};
