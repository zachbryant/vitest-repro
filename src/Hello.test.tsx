import { vi, it, expect } from "vitest";
import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import { Hello } from "./Hello";

vi.mock("./def.tsx", () => ({
  Def: () => <div>Def</div>,
  default: () => <div>Should not be called default in snapshot</div>,
}));

it("matches snapshot", () => {
  const shallowRender = ShallowRenderer.createRenderer();
  const result = shallowRender.render(<Hello />);
  expect(result).toMatchSnapshot();
});
