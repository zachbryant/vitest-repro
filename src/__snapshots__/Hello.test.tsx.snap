// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  style={
    {
      "textAlign": "center",
    }
  }
>
  <div>
    light
  </div>
  <default />
</div>
`;
